### PR TITLE
fix(auth): wallet password validation in creation and update flows

### PR DESCRIPTION
**Pull request by takenagain**
Closes https://github.com/KomodoPlatform/komodo-wallet/issues/2612

Test for wallet creation and settings > update password.

- [ ] Password can't contain the word password
- [ ] Password length should be at least 8 characters long
- [ ] Password should contain at least 1 digit
- [ ] Password should contain at least 1 lowercase character
- [ ] Password should contain at least 1 uppercase character
- [ ] Password should contain at least 1 special character
- [ ] Password can't contain the same character 3 times in a row
- [ ] Settings toggle for `allow weak password` available before login
- [ ] When `allow weak password` is enabled, above constraints not enforced
- [ ] Test non-standard chars like emojis, Devanagari script, and vowels with diaeresis/umlaut etc. to confirm [length constraints](https://github.com/KomodoPlatform/komodo-wallet/pull/2620/files#diff-972e412ec569d4c313a7878ca1efc0874c07b8af8a4b3da1b25b4b3d63681f88R67-R72) are enforced.

## Changes

- Updates the password validation to match KDF password policies
- Adds a toggle for the `allow_weak_password` config parameter for support/debugging in the General settings menu. 
- Updates the SDK commit hash to `95191579e721eaedfd536c747991e6aa7ff1f4dc`

## Summary by CodeRabbit

- **New Features**
  - Added an option in settings to allow weak passwords, including a toggle and explanatory description.
  - Password validation now provides detailed feedback for specific requirements (length, digits, uppercase, lowercase, special character, consecutive characters, and forbidden words).
  - Settings section now includes a "Password Security" label and management UI.

- **Bug Fixes**
  - Improved password validation error messages for clarity and specificity.

- **Tests**
  - Significantly expanded and improved password validation test coverage, including Unicode and edge cases.

- **Style**
  - Updated password validation logic for consistency and clarity in user feedback.

### CODE REVIEW

**Comment by takenagain** [comment]
@coderabbitai review

File: lib/shared/utils/validators.dart [hunk 1]
```diff
@@ -1,15 +1,107 @@
...
+
+  // Check for uppercase (any uppercase letter in any script)
+  if (!RegExp(r'.*\p{Lu}.*', unicode: true).hasMatch(password)) {
+    return PasswordValidationError.missingUppercase;
+  }
+
+  // Check for special characters
+  if (!RegExp(r'.*[^\p{L}\p{N}].*', unicode: true).hasMatch(password)) {
+    return PasswordValidationError.missingSpecialCharacter;
+  }
```
[the remark this answers is not in the document]
Comment (takenagain) [reply lib/shared/utils/validators.dart:95]: Dart RegExp supports Unicode-property escapes from version 2.4: https://api.dart.dev/dart-core/RegExp/isUnicode.html#:~:text=In%20Unicode%20mode%2C%20Dart%20restricts,allowed%20in%20non%2DUnicode%20mode.

File: lib/shared/utils/validators.dart [hunk 2]
```diff
@@ -1,15 +1,107 @@
...
+/// - No same character 3 times in a row
+String? validatePassword(String password) {
+  return passwordErrorMessage(checkPasswordRequirements(password));
+}
+
+/// Internal validation method that returns the enum error type
+PasswordValidationError checkPasswordRequirements(String password) {
+  if (password.length < 8) {
+    return PasswordValidationError.tooShort;
+  }
```
[the remark this answers is not in the document]
Comment (takenagain) [reply lib/shared/utils/validators.dart]: Done in [572a5f2](https://github.com/KomodoPlatform/komodo-wallet/pull/2620/commits/572a5f2b86a19fb8944710a4d3e456c55f0b7050)

File: lib/bloc/auth_bloc/auth_bloc.dart [hunk 3]
```diff
@@ -68,23 +75,27 @@ class AuthBloc extends Bloc<AuthBlocEvent, AuthBlocState> {
         );
       }
 
-      _log.info('login  from a wallet');
+      _log.info('login from a wallet');
       emit(AuthBlocState.loading());
+
+      final weakPasswordsAllowed = await _areWeakPasswordsAllowed();
+
```
[the remark this answers is not in the document]
Comment (takenagain) [reply lib/bloc/auth_bloc/auth_bloc.dart:82]: Done in [29295bd](https://github.com/KomodoPlatform/komodo-wallet/pull/2620/commits/29295bd8615e78fbae99cc843b5e2923a0b556cb)

**Review by smk762 — CHANGES_REQUESTED** [review]
KDF password constraints functioning as expected when `allow weak password` is disabled. :heavy_check_mark: 
When `allow weak password` is enabled, password constraints are still enforced :no_entry_sign: 
Additionally, while `allow weak password` is enabled, all validation error messages are displayed when any of the constraints are not met.

https://github.com/user-attachments/assets/58cf511f-7a2e-4df6-bace-8f9dfa410a38

**Comment by takenagain** [comment]
> When `allow weak password` is enabled, password constraints are still enforced 🚫
> Additionally, while `allow weak password` is enabled, all validation error messages are displayed when any of the constraints are not met.

Fixed in [be23475](https://github.com/KomodoPlatform/komodo-wallet/pull/2620/commits/be23475ad531dd8860ef878cc4c77e8d6135a0b0). All validation checks are now skipped when `allow weak password` is enabled (rather than reverting to the legacy validation, like it did in the video).

**Review by smk762 — APPROVED** [review]
Confirmed functioning as expected on wallet create/import and password change.

**Review by AndrewDelaney — APPROVED** [review]
All seems to be working as intended

**Review by ShantanuSharma9873 — APPROVED** [review]
The functionality is working as expected. Looks good to me !
